airbyte-integrations/connectors/source-google-sheets/integration_tests/integration_test.py

```diff
@@ -0,0 +1,121 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+import pkgutil
+import shutil
+from pathlib import Path
+from typing import Dict
+
+from airbyte_protocol import AirbyteCatalog, ConnectorSpecification
+from apiclient import discovery
+from base_python_test import StandardSourceTestIface
+from google_sheets_source.helpers import Helpers
+from google_sheets_source.models.spreadsheet import Spreadsheet
+
+# Override default permission scopes to allow creating and editing spreadsheets
+SCOPES = [
+    "https://www.googleapis.com/auth/spreadsheets",
+    "https://www.googleapis.com/auth/drive",
+    "https://www.googleapis.com/auth/drive.file",
+]
+
+
+class GoogleSheetsSourceStandardTest(StandardSourceTestIface):
```

Comment:
i like this alot. can you add issues for other features that would have made this easier? seems like the main thing is not having to write files would have been nice in your case.